### DISCUSSION TITLE
ts-node not respecting subfolder tsconfig.json

### DISCUSSION THREAD

**Discussion by moltenice** [TypeStrong/ts-node#1289] [Q&A] [answered]
./tsconfig.json:

```json
{
  "compilerOptions": {
    ...omitted irrelevant config...
    "isolatedModules": true
  },
  "include": [
    "src"
  ],
  "references": [
    {
      "path": "./scripts/tsconfig.json"
    }
  ]
}
```

./scripts/tsconfig.json:

```json
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "isolatedModules": false
  }
}

```

When I run `ts-node scripts/Script.ts` I get this error:

```
'Script.ts' cannot be compiled under '--isolatedModules' because it is considered a global script file. Add an import, export, or an empty 'export {}' statement to make it a module.
```

The question is, why? Why doesn't it respect the fact that `isolatedModules` is disabled for the Scripts folder?

Adding a `--script-mode` fixes it but I want to get it to work without needing to pass anything to `ts-node`.

**Comment by cspotcode** [comment, marked as answer]
Use `ts-node-script`, which enables `--script-mode` by default.  This behavior will become the default for `ts-node` in the next release.
